homeassistant/components/iaqualink/sensor.py

```diff
@@ -36,18 +35,19 @@ def name(self) -> str:
     @property
     def unit_of_measurement(self) -> str:
         """Return the measurement unit for the sensor."""
-        if self.dev.system.temp_unit == "F":
-            return TEMP_FAHRENHEIT
-        return TEMP_CELSIUS
+        if self.dev.name.endswith("_temp"):
+            if self.dev.system.temp_unit == "F":
+                return TEMP_FAHRENHEIT
+            return TEMP_CELSIUS
 
     @property
     def state(self) -> str:
         """Return the state of the sensor."""
-        return int(self.dev.state) if self.dev.state != "" else None
-
-    @property
-    def device_class(self) -> Optional[str]:
```

Comment:
Why do we remove device_class?

Reply:
In my testing, device class seems to be inferred from the unit of measurement. I didn't check the code however.

Reply:
You can see it below. It's the same test as you added above to unit of measurement. We should keep the device_class property.

Reply:
Looking at the code, I didn't see any smarts to infer device_class from unit_of _measurement so I'm guessing the frontend code is smarter in that respect since it showed the thermometer icon just fine.

I've re-added device_class as suggested.